apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/EphemeralTransaction.cs

```diff
@@ -133,13 +133,14 @@ async Task<int> Impl(SpannerTransaction transaction)
             }
         }
 
-        public Task<ReliableStreamReader> ExecuteReadOrQueryAsync(ReadOrQueryRequest request, CancellationToken cancellationToken, int timeoutSeconds /* ignored */)
+        public Task<ReliableStreamReader> ExecuteReadOrQueryAsync(ReadOrQueryRequest request, CancellationToken cancellationToken, int timeoutSeconds /* ignored */, bool isDmlReturn = false)
         {
             return ExecuteHelper.WithErrorTranslationAndProfiling(Impl, "EphemeralTransaction.ExecuteReadOrQuery", _connection.Logger);
 
             async Task<ReliableStreamReader> Impl()
             {
-                PooledSession session = await _connection.AcquireSessionAsync(null, cancellationToken).ConfigureAwait(false);
+                var transactionOptions = isDmlReturn ? SpannerConnection.ReadWriteTransactionOptions : null;
```

Comment:
This means it's not automatically retried - whereas ExecuteNonQuery would use RunWithRetriableTransactionAsync. I suspect we'd want consistency there.